src/Host/Client/Impl/Host/BrokerClient.cs

```diff
@@ -87,10 +88,18 @@ internal abstract class BrokerClient : IBrokerClient {
         protected abstract void OnCredentialsValidated(bool isValid);
 
         public async Task PingAsync() {
-            try {
-                (await HttpClient.PostAsync("/ping", new StringContent(""))).EnsureSuccessStatusCode();
-            } catch (HttpRequestException ex) {
-                throw new RHostDisconnectedException(Resources.Error_HostNotResponding.FormatInvariant(ex.Message), ex);
+            if (HttpClient != null) {
+                var isOnline = await CheckMachineOnlineAsync();
```

Comment:
Sure, we don't need to ping local client

---
In reply to: [81588525](https://github.com/Microsoft/RTVS/pull/2416#discussion_r81588525) [](ancestors = 81588525,81588299,81587401,81587164)